File: 1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */
#include "map"
#include "vector"
using namespace std;

// @lc code=start
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        vector<int> res;
        map<int, int> numMap;
        auto it = nums.begin();
        for (int i = 0; i < nums.size(); i++)
        {
            int ele = nums[i];
            if (numMap.find(target - ele) != numMap.end())
            {
                res.push_back(numMap.find(target - ele)->second);
                res.push_back(i);
                break;
            }
            numMap.insert(pair<int, int>(nums[i], i));
        }
        return res;
    }
};
// @lc code=end

File: 1.两数之和.java

/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */
import java.util.Map;
import java.util.TreeMap;

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = { -1, -1 };
        Map<Integer, Integer> num2idx = new TreeMap();
        for (int i = 0; i < nums.length; i++) {
            if (num2idx.containsKey(target - nums[i])) {
                res[0] = num2idx.get(target - nums[i]);
                res[1] = i;
            } else {
                num2idx.put(nums[i], i);
            }
        }
        return res;
    }
}
// @lc code=end
